client/state/selectors/can-site-view-atomic-hosting.js

```diff
@@ -20,9 +23,21 @@ export default function canSiteViewAtomicHosting( state ) {
 
 	const siteId = getSelectedSiteId( state );
 
-	if ( ! isSiteOnAtomicPlan( state, siteId ) ) {
+	if ( ! canCurrentUser( state, siteId, 'manage_options' ) ) {
+		return false;
+	}
+
+	if ( false !== isVipSite( state, siteId ) ) {
 		return false;
 	}
 
-	return canCurrentUser( state, siteId, 'manage_options' );
+	if ( false !== isJetpackSite( state, siteId ) && true !== isSiteAtomic( state, siteId ) ) {
```

Comment:
I had to read the line a couple of times to understand this condition 😅. Is there any reason why this cannot be simplified as below?
```suggestion
	if ( isJetpackSite( state, siteId ) && ! isSiteAtomic( state, siteId ) ) {
```

Reply:
> Is there any reason why this cannot be simplified as below?

These selectors can return `null` if the site being a Jetpack site is undetermined. With `null` being falsey, an `isJetpackSite` check will evaluate `false` if it's unknown if the site is a Jetpack site, and continue instead of bailing. I only wanted to allow continuing if we're certain the site is not a Jetpack site, hence the awkward direct comparison of `false`. You're right, it's horrible looking and mind-bending, but it ensures that only sites we _know_ are Atomic sites will continue (same thing with VIP).

Reply:
What if you added a condition for `! site` at the top and didn't rely on looking for `null`s here? that might be more declarative and ensure nothing slips through.

```
const site = getSite( state, siteId );
if ( ! site ) {
     return false;
}
```

Reply:
Or could also just add an extra condition with lodash `isNull`